src/rez/package_resources.py

```diff
@@ -303,6 +304,11 @@ def _uri(self):
         """
         raise NotImplementedError
 
+    def install(self, location):
```

Comment:
Should this even be here? Shouldn't it be the responsibility of the ArtifactRepository to know how to install a resource? Wouldn't this break if we're using s3 as an artifact repository since it can't just run a basic copytree from s3?